### PR TITLE
Fix #1587: Refactor isOpenshift method in OpenshiftAdapterSupport in order to be compatible with Openshift 4

### LINKED ISSUE #1587: OpenShiftAdapter uses deprecated APIs to check if cluster is an OpenShift cluster
The OpenShift DevEx team has identified two issues with the `OpenShiftAdapter.isOpenShift` [1] method used by kubernetes client:

1. It checks the `/oapi/` path, which has been deprecated in 4.1
2. It starts walking from the root path`/` to see if the apiserver supports OpenShift components. The OpenShift auth team wants to prevent discovery at the root level - this blocks their effort [2].

[1] https://github.com/fabric8io/kubernetes-client/blob/master/openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftAdapterSupport.java#L48-L76
[2] https://github.com/openshift/origin/pull/23049

### PR DESCRIPTION
**Pull request by rohanKanojia**
follow up PR for #1640

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@gabemontero : Could you please review?

File: openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftAdapterSupport.java [hunk 1]
```diff
@@ -16,63 +16,68 @@
...
+      try {
+        Request.Builder requestBuilder = new Request.Builder()
+          .get()
+          .url(URLUtils.join(masterUrl.toString(), APIS));
+        Response response = httpClient.newCall(requestBuilder.build()).execute();
+        ObjectMapper objectMapper = new ObjectMapper();
+        APIGroupList apiGroupList = objectMapper.readValue(response.body().string(), APIGroupList.class);
+
+        for (APIGroup apiGroup : apiGroupList.getGroups()) {
+          if (apiGroup.getName().endsWith("openshift.io")) {
```
Comment (gabemontero) [inline openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftAdapterSupport.java]: @rohanKanojia - ok, based on what I see with https://github.com/fabric8io/kubernetes-client/pull/1640/files, the `apiGroup.getName()` seems like the correct method to call to get the values that could end with `openshift.io`.

I don't see the actual generated APIGroup and APIGroupList java files in the repo yet.  I'm assuming those get generated as part of either merging this PR or some subsequent task?

Unless the actually generated java/class file unveils some sort of surprise, I think this looks good.

Comment (akram) [reply openshift-client/src/main/java/io/fabric8/openshift/client/OpenshiftAdapterSupport.java]: Hi @gabemontero and @rohanKanojia 

it looks like the code to generate them has been changed here https://github.com/fabric8io/kubernetes-client/commit/2b29f43ca2314df7c3f93424420cf170c4f5017d#diff-4e2a515b790160e4becee6719409c1d1R70 

and maybe it was in a different PR.

@rohanKanojia could you please remember to put me also on CC with @waveywaves ?
we are handing over the Jenkins plugin code base from @gabemontero

**Comment by rohanKanojia** [comment]
retest this please

**Comment by rohanKanojia** [comment]
I'm merging this

**Pull request merged**